doc/troubleshooting.txt

```diff
@@ -0,0 +1,368 @@
+
+This error message cannot be made more explicit because the relevant aspects of Python's
+internals cannot be modified.
+
+.. _out_of_memory:
+
+How to solve Out of memory Error
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Occasionally Theano may fail to allocate memory when there appears to be more
+than enough reporting:
+
+    Error allocating X bytes of device memory (out of memory). Driver report Y
+    bytes free and Z total.
+
+where X is far less than Y and Z (i.e. X << Y < Z).
+
+This scenario arises when an operation requires allocation of a large contiguous
+block of memory but no blocks of sufficient size are available.
+
+GPUs do not have virtual memory and as such all allocations must be assigned to
+a continuous memory region. CPUs do not have this limitation because or their
+support for virtual memory. Multiple allocations on a GPU can result in memory
+fragmentation which can makes it more difficult to find contiguous regions
+of memory of sufficient size during subsequent memory allocations.
+
+A known example is related to writing data to shared variables. When updating a
+shared variable Theano will allocate new space if the size of the data does not
+match the size of the space already assigned to the variable. This can lead to
+memory fragmentation which means that a continugous block of memory of
+sufficient capacity may not be available even if the free memory overall is
+large enough.
+
+.. _float64_output:
+
+theano.function returns a float64 when the inputs are float32 and int{32, 64}
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It should be noted that using float32 and int{32, 64} together
+inside a function would provide float64 as output.
+
+Since the GPU can't compute this kind of output, it would be
+preferable not to use those dtypes together.
+
+To help you find where float64 are created, see the
+:attr:`warn_float64` Theano flag.
+
+.. _test_theano:
+
+How to test that Theano works properly
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An easy way to check something that could be wrong is by making sure ``THEANO_FLAGS``
+have the desired values as well as the ``~/.theanorc``
+
+Also, check the following outputs :
+
+.. code-block:: bash
+
+    ipython
+
+.. code-block:: python
+
+    import theano
+    theano.__file__
+    theano.__version__
+
+
+Once you have installed Theano, you should run the test suite.
+ 
+.. code-block:: bash
+ 
+    python -c "import numpy; numpy.test()"
+    python -c "import scipy; scipy.test()"
+    THEANO_FLAGS=''; python -c "import theano; theano.test()"
+ 
+All Theano tests should pass (skipped tests and known failures are normal). If
+some test fails on your machine, you are encouraged to tell us what went
+wrong on the ``theano-users@googlegroups.com`` mailing list.
+
+.. warning::
+    Theano's test should **NOT** be run with `device=gpu` or they will fail.
```

Comment:
Done. I also specified to use `force_device` to prevent the tests from using the gpu.

Side note: apparently Sphinx does not allow cross-references to `:attr:`, so I cannot link directly to the appropriate section in the configuration page.